File: src/controllers/tokenHealth.ts
import { Context } from "hono";
import { refreshPixivAccessToken } from "../clients/pixiv";

export async function tokenHealthController(c: Context) {
  const refreshToken = process.env.PIXIV_REFRESH_TOKEN;
  if (!refreshToken) {
    return c.json(
      {
        ok: false,
        valid: false,
        error: "Missing env: PIXIV_REFRESH_TOKEN",
      },
      500,
    );
  }

  try {
    await refreshPixivAccessToken(refreshToken);
    return c.json({
      ok: true,
      valid: true,
      message: "Refresh token is valid.",
    });
  } catch (error) {
    return c.json(
      {
        ok: false,
        valid: false,
        error: "Refresh token is invalid or expired.",
        detail: (error as Error).message,
      },
      500,
    );
  }
}

File: controller/tokenHealth.go
package controller

import (
	"github.com/gofiber/fiber/v2"
	"pixivhono/client"
	"pixivhono/config"
)

func TokenHealthController(c *fiber.Ctx) error {
	refreshToken := config.GlobalConfig.PixivRefreshToken
	if refreshToken == "" {
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"ok":    false,
			"valid": false,
			"error": "Missing env: PIXIV_REFRESH_TOKEN",
		})
	}

	_, err := client.RefreshPixivAccessToken(refreshToken)
	if err != nil {
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"ok":     false,
			"valid":  false,
			"error":  "Refresh token is invalid or expired.",
			"detail": err.Error(),
		})
	}

	return c.JSON(fiber.Map{
		"ok":      true,
		"valid":   true,
		"message": "Refresh token is valid.",
	})
}
